tegra-fuse.c

```diff
@@ -173,26 +183,36 @@ tegra_fuse_context_open (const char *basepath)
 	unsigned long chipid;
 	char soctype[65];
 
-	fd = open("/sys/module/tegra_fuse/parameters/tegra_chip_id", O_RDONLY);
-	if (fd < 0)
-		return NULL;
+	fd = open("/sys/module/tegra_fuse/parameters/tegra_chip_uid", O_RDONLY);
+	if (fd < 0) {
+		fd = open("/sys/module/fuse_burn/parameters/tegra_chip_uid", O_RDONLY);
+		if (fd < 0)
+			return NULL;
+	}
+
 	typelen = read(fd, soctype, sizeof(soctype)-1);
 	close(fd);
 	if (typelen < 0)
 		return NULL;
 	while (typelen > 0 && soctype[typelen-1] == '\n') typelen--;
 	soctype[typelen] = '\0';
+
 	chipid = strtoul(soctype, NULL, 10);
+	chipid = (chipid >> 60);
```

Comment:
This only applies for the new-style case, and you're not handling the old-style case anywhere

Reply:
Please check my comment above